src/cdk-experimental/radio/radio.ts

```diff
@@ -0,0 +1,178 @@
+ *   <label cdkRadioButton value="2">Option 2</label>
+ *   <label cdkRadioButton value="3">Option 3</label>
+ * </div>
+ * ```
+ */
+@Directive({
+  selector: '[cdkRadioGroup]',
+  exportAs: 'cdkRadioGroup',
+  host: {
+    'role': 'radiogroup',
+    'class': 'cdk-radio-group',
+    '[attr.tabindex]': 'pattern.tabindex()',
+    '[attr.aria-readonly]': 'pattern.readonly()',
+    '[attr.aria-disabled]': 'pattern.disabled()',
+    '[attr.aria-orientation]': 'pattern.orientation()',
+    '[attr.aria-activedescendant]': 'pattern.activedescendant()',
+    '(keydown)': 'pattern.onKeydown($event)',
+    '(pointerdown)': 'pattern.onPointerdown($event)',
+    '(focusin)': 'onFocus()',
+  },
+})
+export class CdkRadioGroup<V> implements AfterViewInit {
+  /** The directionality (LTR / RTL) context for the application (or a subtree of it). */
+  private readonly _directionality = inject(Directionality);
+
+  /** The CdkRadioButtons nested inside of the CdkRadioGroup. */
+  private readonly _cdkRadioButtons = contentChildren(CdkRadioButton, {descendants: true});
+
+  /** A signal wrapper for directionality. */
+  protected textDirection = toSignal(this._directionality.change, {
+    initialValue: this._directionality.value,
+  });
+
+  /** The RadioButton UIPatterns of the child CdkRadioButtons. */
+  protected items = computed(() => this._cdkRadioButtons().map(radio => radio.pattern));
+
+  /** Whether the radio group is vertically or horizontally oriented. */
+  orientation = input<'vertical' | 'horizontal'>('vertical');
+
+  /** Whether focus should wrap when navigating. */
+  wrap = input(false, {transform: booleanAttribute}); // Radio groups typically don't wrap
+
+  /** Whether disabled items in the group should be skipped when navigating. */
+  skipDisabled = input(true, {transform: booleanAttribute});
+
+  /** The focus strategy used by the radio group. */
+  focusMode = input<'roving' | 'activedescendant'>('roving');
+
+  /** Whether the radio group is disabled. */
+  disabled = input(false, {transform: booleanAttribute});
+
+  /** Whether the radio group is readonly. */
+  readonly = input(false, {transform: booleanAttribute});
+
+  /** The value of the currently selected radio button. */
+  value = model<V | null>(null);
+
+  /** The internal selection state for the radio group. */
+  private readonly _value = linkedSignal(() => (this.value() ? [this.value()!] : []));
+
+  /** The current index that has been navigated to. */
+  activeIndex = model<number>(0);
+
+  /** The RadioGroup UIPattern. */
+  pattern: RadioGroupPattern<V> = new RadioGroupPattern<V>({
+    ...this,
+    items: this.items,
+    value: this._value,
+    textDirection: this.textDirection,
+  });
+
+  /** Whether the radio group has received focus yet. */
+  private _hasFocused = signal(false);
+
+  /** Whether the radio buttons in the group have been initialized. */
+  private _isViewInitialized = signal(false);
+
+  constructor() {
+    effect(() => {
```

Comment:
could this just be `afterRenderEffect` and you skip the `ngAfterViewInit` stuff?

Reply:
I'm just now learning about `afterRenderEffect` but that sounds exactly like what we want haha